bot/exts/utilities/githubinfo.py

```diff
@@ -33,11 +175,66 @@ async def github_group(self, ctx: commands.Context) -> None:
         if ctx.invoked_subcommand is None:
             await invoke_help_command(ctx)
 
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """
+        Automatic issue linking.
+
+        Listener to retrieve issue(s) from a GitHub repository using automatic linking if matching <org>/<repo>#<issue>.
+        """
+        # Ignore bots
+        if message.author.bot:
+            return
+
+        issues = [
+            FoundIssue(*match.group("org", "repo", "number"))
+            for match in AUTOMATIC_REGEX.finditer(self.remove_codeblocks(message.content))
+        ]
+        links = []
+
+        if issues:
+            # Block this from working in DMs
+            if not message.guild:
+                return
+
+            log.trace(f"Found {issues = }")
+            # Remove duplicates
+            issues = set(issues)
```

Comment:
I wonder if there is a better way we can do this, in order to keep the originally provided order, if there's multiple.

Reply:
Yes, we can use a dict and do dict.fromkeys(issues)

Reply:
that sounds so dumb (but it would work!), I wish something like OrderedSet was in the stdlib.

Reply:
Hi, any resolution on what we should do here?

Reply:
part of #968